### PR TITLE
Fix xshard receipt log generation

### PR DESCRIPTION
**Pull request by ninjaahhh**
otherwise may have out of index errors when generating receipts with logs

### CODE REVIEW

File: quarkchain/cluster/filter.py [hunk 1]
```diff
@@ -87,8 +87,15 @@ def _get_logs(self, blocks: List[MinorBlock]) -> List[Log]:
         """Given potential blocks, re-run tx to find exact matches."""
         ret = []
         for b_i, block in enumerate(blocks):
-            for i in range(len(block.tx_list or [])):
-                r = block.get_receipt(self.db.db, i)
+            deposit_hlist = (
```
Comment (qizhou) [inline quarkchain/cluster/filter.py]: I would suggest adding one line comment to illustrate missing case (old version db)

File: quarkchain/cluster/shard_state.py [hunk 2]
```diff
@@ -1548,7 +1548,13 @@ def get_transaction_receipt(
         if not block:
             return None
         try:
-            receipt = block.get_receipt(self.evm_state.db, index)
+            deposit_hlist = None
+            if index >= len(block.tx_list):
+                deposit_hlist = self.db.get_xshard_deposit_hash_list(
```
Comment (qizhou) [inline quarkchain/cluster/shard_state.py]: deposit_hlist may be None

File: quarkchain/cluster/filter.py (hunk 1 above)
Comment (ninjaahhh) [reply quarkchain/cluster/filter.py]: done. put the older version db handling into the `block. get_receipt`, by returning a fake tx receipt

File: quarkchain/cluster/shard_state.py (hunk 2 above)
Comment (ninjaahhh) [reply quarkchain/cluster/shard_state.py]: done. similar to previous note

**Pull request merged**